### PR TITLE
feat(universal): add abstract classes for Title, Cookie and LocalStorage (wip)

### PR DESCRIPTION
**Pull request by manekinekko**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/angular/universal/blob/master/CONTRIBUTING.md#commit-message-format
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

* **What modules are related to this pull-request**
- [ ] express-engine
- [ ] grunt-prerender
- [ ] gulp-prerender
- [ ] hapi-engine
- [ ] preboot
- [ ] universal-preview
- [x] universal
- [ ] webpack-prerender

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
remove unused proxy imports from some test files

Initial implementation of an abstract classes for `Title`, `Cookie` and `LocalStorage`... (WIP)

### CODE REVIEW

**Comment by JohnGorter** [comment]
Hey Wassim, 

Your class has the name TitleAdapater... Shouldn't that be TitleAdapter :-)...

**Comment by manekinekko** [comment]
Haha! Of course. Thanks :)

File: modules/universal/src/node/platform/localStorage.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+/**
+ * The Storage abstract class of the Web Storage API provides access to local storage for a particular
+ * domain.
+ */
+abstract class LocalStorageAdapater {
```
Comment (manekinekko) [inline modules/universal/src/node/platform/localStorage.ts]: LocalStorage~~Adapater~~ → LocalStorage**Adapter**

File: modules/universal/src/node/platform/title.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+/**
+ * An abstract service that can be used to get and set the title of a current HTML document.
+ */
+abstract class TitleAdapater {
```
Comment (manekinekko) [inline modules/universal/src/node/platform/title.ts]: Title~~Adapater~~ → Title**Adapter**

**Comment by gdi2290** [comment]
@manekinekko can you remove Adapter and make the set/get async with Observable return type and getValue/setValue the sync call

**Comment by manekinekko** [comment]
@gdi2290 I'll refactor the code

**Comment by joaogarin** [comment]
Hello @manekinekko while this is still a WIP is there a way to set the page title using universal? not very important for now, more on a curiosity note as to how it could work.

**Comment by gdi2290** [comment]
I'm going to merge this then refactor

**Pull request merged**